datafusion/physical-expr/src/aggregate/sum.rs

```diff
@@ -338,6 +455,42 @@ impl Accumulator for SumAccumulator {
     }
 }
 
+#[derive(Debug)]
+struct SumAccumulatorV2 {
+    index: usize,
+    datatype: DataType,
+}
+
+impl SumAccumulatorV2 {
+    pub fn new(index: usize, datatype: DataType) -> Self {
+        Self { index, datatype }
+    }
+}
+
+impl AccumulatorV2 for SumAccumulatorV2 {
+    fn update_batch(
+        &mut self,
+        values: &[ArrayRef],
+        accessor: &mut RowAccessor,
+    ) -> Result<()> {
+        let values = &values[0];
+        add_to_row(&self.datatype, self.index, accessor, &sum_batch(values)?)?;
```

Comment:
I wonder if it is needed to go through `sum_batch` here (which turns the sum into a `ScalarValue`) -- perhaps we could call the appropriate sum kernel followed by  a direct update

Reply:
I initially use `sum_batch` here mainly to reduce code duplication with that of `SumAccumulator`, besides, there's a decimal `sum_batch` that isn't included in the compute kernel yet.

Reply:
Also, possibly related https://github.com/apache/arrow-datafusion/issues/2447